sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java

```diff
@@ -414,15 +414,23 @@ private boolean shouldOwnMorePartitions(final int minPartitionsPerEventProcessor
     private Map<String, PartitionOwnership>
         removeInactivePartitionOwnerships(final Map<String, PartitionOwnership> partitionOwnershipMap) {
         return partitionOwnershipMap.entrySet().stream().filter(entry -> {
-            long diff = (System.currentTimeMillis() - entry.getValue().getLastModifiedTime()) / 1000;
-            LOGGER.atLevel((diff < inactiveTimeLimitInMillis) ? LogLevel.VERBOSE : LogLevel.INFORMATIONAL)
+
+            final long nowInMillis = System.currentTimeMillis();
+            final long lastModifiedTimeInMillis = entry.getValue().getLastModifiedTime();
+            long diffInMillis = (nowInMillis - lastModifiedTimeInMillis);
+            final boolean isActive = (diffInMillis < inactiveTimeLimitInMillis);
+            final LogLevel logLevel = isActive ? LogLevel.VERBOSE : LogLevel.INFORMATIONAL;
+
+            LOGGER.atLevel(logLevel)
                 .addKeyValue(PARTITION_ID_KEY, entry.getKey())
                 .addKeyValue(OWNER_ID_KEY, ownerId)
                 .addKeyValue("partitionOwnerId", entry.getValue().getOwnerId())
-                .addKeyValue("modifiedSecondsAgo", diff)
+                .addKeyValue("lastModifiedTime", lastModifiedTimeInMillis)
+                .addKeyValue("modifiedSecondsAgo", diffInMillis / 1000)
```

Comment:
nit

```suggestion
                .addKeyValue("modifiedSecondsAgo", diffInMillis / 1000d)
```

in case fractions matter?

Reply:
thanks, updated to use this.